src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/ITransaction.cs

```diff
@@ -216,6 +216,13 @@ public interface ITransaction
         /// <exception cref="ArgumentNullException"></exception>
         void SetCustomTransactionName(string name, TransactionNamePriority priority = TransactionNamePriority.Uri);
 
+        /// <summary>
+        /// Set the Request Method for the current transaction (if there is one).
+        /// </summary>
+        /// <param name="requestMethod">The Request Method for this transaction. Must not be null.</param>
+        /// <exception cref="ArgumentNullException"></exception>
```

Comment:
are we sure it throws? I don't see parameter validation anywhere

Reply:
Happens in the implementation here:
https://github.com/newrelic/newrelic-dotnet-agent/blob/ec3a5800a2d9844f07243e8c919294d059c2b901/src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs#L778-L781